src/sql-parser/src/ast/defs/value.rs

```diff
@@ -179,23 +179,31 @@ mod test {
 pub enum DataType {
     /// Array
     Array(Box<DataType>),
-    /// Fixed-length character type e.g. CHAR(10)
-    Char(Option<u64>),
-    /// Decimal type with optional precision and scale e.g. DECIMAL(10,2)
-    Decimal(Option<u64>, Option<u64>),
-    /// Floating point with optional precision e.g. FLOAT(8)
-    Float(Option<u64>),
     /// List
     List(Box<DataType>),
     /// Map
     Map {
         key_type: Box<DataType>,
         value_type: Box<DataType>,
     },
-    /// Types whose names don't accept parameters, e.g. INT
-    Other(Ident),
-    /// Variable-length character type e.g. VARCHAR(10)
-    Varchar(Option<u64>),
+    /// Types who don't embed other types, e.g. INT
+    Other { name: ObjectName, typ_mod: Vec<u64> },
+}
+
+impl DataType {
+    pub fn internal_canonical_name(&self) -> Option<ObjectName> {
+        if let DataType::Other { name, .. } = self {
+            let canonical = |n: &str| Some(ObjectName(vec![Ident::new(n)]));
```

Comment:
Thought separate from these changes, I feel like I've written things like this a bunch:
```
ObjectName(vec![Ident::new(n)])
```
Maybe we should make a util function for it!

Reply:
Added `From<&str> for ObjectName`--there was a surprising number of them. We'll see if this passes muster or if it should just have a dedicated function.